### PR TITLE
Fix/remove transfer send

### PR DESCRIPTION
**Pull request by Tristan22400**
Fix: https://github.com/morpho-labs/hack-smart-contracts/issues/1

### CODE REVIEW

File: src/Hack.sol [hunk 1]
```diff
@@ -84,9 +85,15 @@ contract HeadOrTail {
...
-        else lastParty.transfer(2 ether);
+        bool success;
 
+        if (_guessHead == lastChoiceHead) {
+            (success,) = msg.sender.call{value: 2 ether}("");
+        } else {
+            (success,) = lastParty.call{value: 2 ether}("");
+        }
+
+        require(success, "Unsuccessful send");
```
Comment (makcandrov) [inline src/Hack.sol:96]: 'Transfer failed" sounds better imo

**Review by makcandrov — APPROVED** [review]

**Review by makcandrov — APPROVED** [review]

File: src/Hack.sol [hunk 2]
```diff
@@ -84,9 +85,15 @@ contract HeadOrTail {
...
 
-        if (_guessHead == lastChoiceHead) payable(msg.sender).transfer(2 ether);
-        else lastParty.transfer(2 ether);
+        bool success;
 
+        if (_guessHead == lastChoiceHead) {
+            (success,) = msg.sender.call{value: 2 ether}("");
+        } else {
+            (success,) = lastParty.call{value: 2 ether}("");
+        }
```
Comment (MerlinEgalite) [inline src/Hack.sol:94]:
```suggestion
     (bool success,) = _guessHead == lastChoiceHead ?
          msg.sender.call{value: 2 ether}("") :
          lastParty.call{value: 2 ether}("");
```

File: src/Hack.sol [hunk 3]
```diff
@@ -583,8 +603,8 @@ contract GuessTheAverage {
     function distribute(uint256 _count) public {
         require(currentStage == Stage.WinnersFound, "Winners must have been found");
         for (uint256 i = cursorDistribute; i < winners.length && (_count == 0 || i < cursorDistribute + _count); i++) {
-            // Send ether to the winners, use send not to block.
-            payable(winners[i]).send(totalBalance / (winners.length - numberOfLosers));
+            // Send ether to the winners, use send not to block, if one of the account cannot receive ETH.
```
Comment (MerlinEgalite) [inline src/Hack.sol:606]: should we write it?

**Review by MerlinEgalite — APPROVED** [review]

File: src/Hack.sol (hunk 3 above)
Comment (makcandrov) [reply src/Hack.sol]: I think it's ok to keep it, so that they don't focus on that for too long

**Review by MerlinEgalite — APPROVED** [review]

**Pull request merged**